File: TAREA 03/Ejercicio4.py
# Solicitamos los dos números enteros al usuario
a = int(input("Ingrese el primer número (a): "))
b = int(input("Ingrese el segundo número (b): "))

# Guardamos los valores originales para el mensaje final
original_a = a
original_b = b

# El bucle se ejecuta mientras b sea diferente de 0
while b != 0:
    # Asignación simultánea: a toma el valor de b, y b toma el residuo de a % b
    a, b = b, a % b

# Al terminar el ciclo, el MCD queda almacenado en la variable 'a'
print(f"El MCD de {original_a} y {original_b} es: {a}")

File: TAREA 03/Ejercicio4.cpp
#include <iostream>
using namespace std;

int main() {
    int a, b;
    cout << "Ingrese el primer numero (a): ";
    cin >> a;
    cout << "Ingrese el segundo numero (b): ";
    cin >> b;

    // Guardamos los valores originales para mostrarlos al final
    int original_a = a;
    int original_b = b;

    // El algoritmo de Euclides dice que repetimos mientras b no sea 0
    while (b != 0) {
        int temporal = b; // Guardamos el valor actual de b
        b = a % b;        // b pasa a ser el residuo de a dividido entre b
        a = temporal;     // a toma el antiguo valor de b
    }

    // Al romperse el ciclo, el MCD queda almacenado en la variable 'a'
    cout << "El MCD de " << original_a << " y " << original_b << " es: " << a << endl;

    return 0;
}

